### PR TITLE
UTs pass1

### PR DESCRIPTION
**Pull request by bharathsreekanth**
# Description
First pass at unit tests for missing files. 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ func (c *Client) GetVersion(ctx context.Context) (string, error) {
 		return "", errNilReponse
 	case resp.StatusCode == http.StatusUnauthorized:
 		// Authenticate then try again
-		if _, err = c.Authenticate(ctx, c.configConnect); err != nil {
+		if _, err = c.Authenticate(ctx, c.configConnect); err == nil {
```
Comment (falfaroc) [inline api.go:91]: **question:** Why the change? We would now be returning `nil`.

**Review by falfaroc — APPROVED** [review]

File: api.go (hunk 1 above)
Comment (bharathsreekanth) [reply api.go:91]: checking, not sure why I did that. or was it not intended.
